app/adapters/use-eoa-txs.ts

```diff
@@ -44,21 +42,9 @@ export function useEOATxs(): WalletAdapter {
     return await comethMarketplaceClient.order.createOrder(buyOffer)
   }
 
-  async function cancelOrder({ nonce, signer }: CancelOrderParams) {
-    const signedPrefix = await signer!.signMessage(`Nonce: ${nonce}`)
-    const signature = splitSignature(signedPrefix)
-    const { r, s, v } = signature
-
-    const body: CancelOrderRequest = {
-      signature: {
-        signatureType: 2,
-        r,
-        s,
-        v,
-      },
-    }
-
-    return await comethMarketplaceClient.order.cancelOrder(nonce, body)
+  async function cancelOrder({ nonce, nftSwapSdk }: CancelOrderParams) {
+    const tx = await nftSwapSdk?.cancelOrder(nonce, "ERC721")
+    return await tx?.wait()
```

Comment:
This function already exists here: https://github.com/cometh-game/marketplace-ui/blob/4120648566db7aa44d997b08a983efa3ebe3cde3/app/adapters/use-cometh-txs.ts#L23 
If needed you can move it into a more generic file, but you should share the same code.